### DISCUSSION TITLE
DJL Monthly Update - November 2021

### DISCUSSION THREAD

**Discussion by zachgk** [deepjavalibrary/djl#1401] [News]
### [DJL](https://github.com/deepjavalibrary/djl)

* [Create TestRequirements utility #1330](https://github.com/deepjavalibrary/djl/pull/1330)
* [Add debugging toString to Input and Output #1327](https://github.com/deepjavalibrary/djl/pull/1327)
* **[Add support for inferentia #1329](https://github.com/deepjavalibrary/djl/pull/1329)**
* [Add support for downloading .mar file #1338](https://github.com/deepjavalibrary/djl/pull/1338)
* [Make BufferedImageFactory.fromNDArray synchronous #1339](https://github.com/deepjavalibrary/djl/pull/1339)
* **[Add opencv extension #1331](https://github.com/deepjavalibrary/djl/pull/1331)**
* [LayerNorm for MXNet #1342](https://github.com/deepjavalibrary/djl/pull/1342)
* [Predictor with device #1346](https://github.com/deepjavalibrary/djl/pull/1346)
* [allow create boolean NDArray #1350](https://github.com/deepjavalibrary/djl/pull/1350)
* [Add codeowners #1356](https://github.com/deepjavalibrary/djl/pull/1356)
* [Adds Image.duplicate() back #1355](https://github.com/deepjavalibrary/djl/pull/1355)
* [Upgrade DJL version from 0.13.0 to 0.14.0 #1357](https://github.com/deepjavalibrary/djl/pull/1357)
* [Upgrade snapcraft to use DJL 0.14.0 #1359](https://github.com/deepjavalibrary/djl/pull/1359)
* [Bump up version to 0.15.0 #1361](https://github.com/deepjavalibrary/djl/pull/1361)
* [Expose api from opencv #1367](https://github.com/deepjavalibrary/djl/pull/1367)
* [Improves TrainingResult with toString() print out #1369](https://github.com/deepjavalibrary/djl/pull/1369)
* **[Updates PyTorch engine to 1.10.0 #1366](https://github.com/deepjavalibrary/djl/pull/1366)**
* [Updates TensorFlowLite engine to 2.6.2 #1372](https://github.com/deepjavalibrary/djl/pull/1372)
* **[Removes native-auto packages #1375](https://github.com/deepjavalibrary/djl/pull/1375)**
* [Refactor LibUtils #1378](https://github.com/deepjavalibrary/djl/pull/1378)
* [Add djl version to manifest #1380](https://github.com/deepjavalibrary/djl/pull/1380)
* [Upload all test reports in single task #1383](https://github.com/deepjavalibrary/djl/pull/1383)
* [Update bom #1384](https://github.com/deepjavalibrary/djl/pull/1384)
* [Allows regiter engine at runtime #1386](https://github.com/deepjavalibrary/djl/pull/1386)
* [move pytorch JNI into separate package #1385](https://github.com/deepjavalibrary/djl/pull/1385)

### [DJL Serving](https://github.com/deepjavalibrary/djl-serving)

* [Refactors minMaxWorkers from ModelInfo to WorkerPool #30](https://github.com/deepjavalibrary/djl-serving/pull/30)
* **[Add autoscale for inferentia #31](https://github.com/deepjavalibrary/djl-serving/pull/31)**
* **[Add support to load torchserve model #32](https://github.com/deepjavalibrary/djl-serving/pull/32)**
* [Add WorkLoadManager.unregisterModel #33](https://github.com/deepjavalibrary/djl-serving/pull/33)
* [auto detect number of neuron cores #34](https://github.com/deepjavalibrary/djl-serving/pull/34)
* [Add dockerfile for inferentia #36](https://github.com/deepjavalibrary/djl-serving/pull/36)
* [install dependency if needed #37](https://github.com/deepjavalibrary/djl-serving/pull/37)
* [support nested folder in model archive file #38](https://github.com/deepjavalibrary/djl-serving/pull/38)
* [detect engine for local model store #35](https://github.com/deepjavalibrary/djl-serving/pull/35)
* [Add initial codeowners #39](https://github.com/deepjavalibrary/djl-serving/pull/39)
* [Upgrade DJL version from 0.13.0 to 0.14.0 #40](https://github.com/deepjavalibrary/djl-serving/pull/40)
* [Reduces netty dependencies #41](https://github.com/deepjavalibrary/djl-serving/pull/41)
* [Upgrade to djl 0.15.0 #44](https://github.com/deepjavalibrary/djl-serving/pull/44)

### [DJL](https://github.com/deepjavalibrary/djl)

* [include all engines in example project build.gradle #1344](https://github.com/deepjavalibrary/djl/pull/1344)
* [Update PyTorch engine document #1352](https://github.com/deepjavalibrary/djl/pull/1352)
* [Fix maven build #1354](https://github.com/deepjavalibrary/djl/pull/1354)
* [Fixes rank classification jupyter notebook #1368](https://github.com/deepjavalibrary/djl/pull/1368)
* [Simplify jupyter notebooks with NoBatchifyTranslator #1370](https://github.com/deepjavalibrary/djl/pull/1370)
* [remove mxnet-native-auto from pom.xml #1387](https://github.com/deepjavalibrary/djl/pull/1387)
* [Avoid using symbolic link #1390](https://github.com/deepjavalibrary/djl/pull/1390)

### [DJL Demos](https://github.com/deepjavalibrary/djl-demo)

* **[Add huggingface inferentia serving example #184](https://github.com/deepjavalibrary/djl-demo/pull/184)**
* Update hugginface inferentia README.
* [huggingface] Add examples to deploy on SageMaker
* [huggingface] Add requirements.txt
* [Add initial codeowners #185](https://github.com/deepjavalibrary/djl-demo/pull/185)
* [Upgrade DJL to 0.14.0 #186](https://github.com/deepjavalibrary/djl-demo/pull/186)
* [upgrade neuron SDK to 1.16.1 #187](https://github.com/deepjavalibrary/djl-demo/pull/187)
* [update inferentia document for 0.14.0 #188](https://github.com/deepjavalibrary/djl-demo/pull/188)

### [DJL](https://github.com/deepjavalibrary/djl)

* [Fix jar model url download bug #1336](https://github.com/deepjavalibrary/djl/pull/1336)
* [Fix calculation for SigmoidBinaryCrossEntropyLoss from sigmoid #1345](https://github.com/deepjavalibrary/djl/pull/1345)
* [workaround windows build failure #1347](https://github.com/deepjavalibrary/djl/pull/1347)
* [improve error message #1348](https://github.com/deepjavalibrary/djl/pull/1348)
* [Handle model archive file contains folder name #1349](https://github.com/deepjavalibrary/djl/pull/1349)
* [Fixes 2nd inference throw exception bug #1351](https://github.com/deepjavalibrary/djl/pull/1351)
* [fix pytorch nightly build #1358](https://github.com/deepjavalibrary/djl/pull/1358)
* [Fix pytorch jni nightly build workflow #1362](https://github.com/deepjavalibrary/djl/pull/1362)
* [Add missing actions repository checks #1363](https://github.com/deepjavalibrary/djl/pull/1363)
* [Fix pytorch jni windows build #1364](https://github.com/deepjavalibrary/djl/pull/1364)
* [Workaround codeql build issue #1388](https://github.com/deepjavalibrary/djl/pull/1388)

### [DJL Serving](https://github.com/deepjavalibrary/djl-serving)

* [serving] fix misuse of access log
* Fix access log naming convention issue
* [workaround codeql build issue #43](https://github.com/deepjavalibrary/djl-serving/pull/43)
